libs/wingsdk/src/cloud/api.ts

```diff
@@ -0,0 +1,144 @@
+import { Construct } from "constructs";
+import { Polycons } from "polycons";
+import { FunctionProps } from "./function";
+import { Inflight, Resource } from "../core";
+import { Code } from "../core/inflight";
+
+/**
+ * Global identifier for `Api`.
+ */
+
+export const API_TYPE = "wingsdk.cloud.Api";
+
+/**
+ * Properties for `Api`.
+ */
+
+export interface ApiProps {}
+
+/**
+ * Functionality shared between all `Api` implementations.
+ */
+
+export abstract class ApiBase extends Resource {
+  public readonly stateful = true;
+  constructor(scope: Construct, id: string, props: ApiProps) {
+    super(scope, id);
+
+    if (!scope) {
+      return;
+    }
+
+    props;
+
+    this.display.title = "Api";
+    this.display.description = "A REST API endpoint";
+  }
+  /**
+   * Add a inflight handler to the api for GET requests on the given route.
+   * @param route The route to handle GET requests for.
+   * @param inflight The function to handle the request.
+   * @param props Options for the route.
+   */
+  public abstract get(
+    route: string,
+    inflight: Inflight,
+    props?: ApiGetProps
+  ): void;
+}
+
+/**
+ * Options for Api.get.
+ */
+export interface ApiGetProps extends FunctionProps {}
+
+/**
+ * AWS infrastructure of `cloud.Api`.
+ */
+export class Api extends ApiBase {
+  constructor(scope: Construct, id: string, props: ApiProps = {}) {
+    super(null as any, id, props);
+    return Polycons.newInstance(API_TYPE, scope, id, props) as Api;
+  }
+
+  /** @internal */
+  public _toInflight(): Code {
+    throw new Error("Method not implemented.");
+  }
+
+  /**
+   *  Create a function to handle GET requests to a route.
+   *
+   * @param route
+   * @param inflight
+   * @param props
+   */
+  public get(route: string, inflight: Inflight, props?: ApiGetProps): void {
+    route;
+    inflight;
+    props;
+    throw new Error("Method not implemented.");
+  }
+}
+
+/**
+ * Allowed HTTP methods for a endpoint.
+ */
+export enum HttpMethod {
+  /** Get */
+  GET = "GET",
+  /** Head */
+  HEAD = "HEAD",
+  /** Post */
+  POST = "POST",
+  /** Put */
+  PUT = "PUT",
+  /** Delete */
+  DELETE = "DELETE",
+  /** Connect */
+  CONNECT = "Connect",
+  /** Options */
+  OPTIONS = "OPTIONS",
+  /** Patch */
+  PATCH = "PATCH",
+}
```

Comment:
running `npx nx wing -- compile --target=tf-aws ../../examples/tests/valid/api.w` ind the `wing/apps` folder produces a error 
```
thread '<unnamed>' panicked at 'Expected cloud.HttpMethod to be defined: TypeError { message: "Unknown symbol \"HttpMethod\" in namespace \"cloud\"", span: WingSpan { start: Point { row: 0, column: 0 }, end: Point { row: 0, column: 0 }, start_byte: 0, end_byte: 0, file_id: "" } }', libs/wingc/src/type_check/jsii_importer.rs:132:14
stack backtrace:
wasm://wasm/09bd1d76:1
RuntimeError: unreachable
    at __rust_start_panic (wasm://wasm/09bd1d76:wasm-function[14371]:0x53cd5d)
    at rust_panic (wasm://wasm/09bd1d76:wasm-function[6112]:0x487c9b)
    at std::panicking::rust_panic_with_hook::hc5c73bd02fe928d4 (wasm://wasm/09bd1d76:wasm-function[996]:0x2a9a6d)
    at std::panicking::begin_panic_handler::{{closure}}::hcd8b0196239bb506 (wasm://wasm/09bd1d76:wasm-function[6875]:0x4a780e)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h634ecf1fa70ae144 (wasm://wasm/09bd1d76:wasm-function[13423]:0x535a2f)
    at rust_begin_unwind (wasm://wasm/09bd1d76:wasm-function[10014]:0x4ffca2)
    at core::panicking::panic_fmt::h1024bc10a158814e (wasm://wasm/09bd1d76:wasm-function[12630]:0x52b9c5)
    at core::result::unwrap_failed::h4f10a8b6d25baa65 (wasm://wasm/09bd1d76:wasm-function[7985]:0x4cd4f7)
    at core::result::Result<T,E>::expect::hb29bc82c2258788a (wasm://wasm/09bd1d76:wasm-function[3682]:0x3fcc6d)
    at wingc::type_check::jsii_importer::JsiiImporter::lookup_or_create_type::h4e62f125c2534baf (wasm://wasm/09bd1d76:wasm-function[635]:0x23e9d2)
```

and I have no idea how to fix that. Maybe some missing export? Do I have to register this enum somewhere?
@Chriscbr can you have a look at this?

Reply:
@RaphaelManke ah, I think this is a limitation as the compiler don't support importing enums quite yet. https://github.com/winglang/wing/issues/1484 For now we might have to use strings in the API a placeholder 😅

Reply:
ok i commented it out.